pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -703,46 +682,48 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(clusterName string, apiService *v1.Serv
 
 		glog.V(4).Infof("Pool for listener %s: %s", listener.ID, pool.ID)
 		members, err := getMembersByPoolID(lbaas.network, pool.ID)
-		if err != nil && !isNotFound(err) {
+		if err != nil && !lbaas.IsNotFound(err) {
 			return nil, fmt.Errorf("Error getting pool members %s: %v", pool.ID, err)
 		}
-		for _, node := range nodes {
-			addr, err := nodeAddressForLB(node)
-			if err != nil {
-				if err == ErrNotFound {
-					// Node failure, do not create member
-					glog.Warningf("Failed to create LB pool member for node %s: %v", node.Name, err)
-					continue
-				} else {
-					return nil, fmt.Errorf("Error getting address for node %s: %v", node.Name, err)
+		if !lbaas.IsNotFound(err) && members != nil {
```

Comment:
Why this ?  if members != nil and of course err returned is nil and do we really need this?